lib/segment/src/payload_storage/simple_payload_storage_impl.rs

```diff
@@ -23,6 +23,21 @@ impl PayloadStorage for SimplePayloadStorage {
         Ok(())
     }
 
+    fn assign_by_key(
+        &mut self,
+        point_id: PointOffsetType,
+        payload: &Payload,
+        key: &str,
+    ) -> OperationResult<()> {
+        match self.payload.get_mut(&point_id) {
+            Some(point_payload) => point_payload.merge_by_key(payload, key)?,
+            None => {
+                self.payload.insert(point_id, payload.to_owned());
```

Comment:
This logic path ignores `key`, that will create a situation where idempotency will not be satisfied.

Reply:
> I plan to return error in this branch

no, it is a valid request. You just need to merge empty payload with given

Reply:
same as we would expect if stored payload doesn't have `key1.key2`